# Generated by Django 4.1.4 on 2023-01-10 11:12

from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    dependencies = [
        ('api', '0002_remove_post_dislikes_alter_user_is_admin'),
    ]

    operations = [
        migrations.CreateModel(
            name='Challenge',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('title', models.CharField(blank=True, default='', max_length=200, null=True)),
                ('description', models.CharField(blank=True, default='', max_length=2000, null=True)),
                ('date', models.DateTimeField(auto_now_add=True, null=True)),
                ('difficulty', models.IntegerField(blank=True, default=1, null=True)),
                ('answer', models.IntegerField(blank=True, default=0, null=True)),
                ('author', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, to='api.user')),
                ('space', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, to='api.space')),
                ('users_that_attempted', models.ManyToManyField(blank=True, related_name='attempted', to='api.user')),
                ('users_that_failed', models.ManyToManyField(blank=True, related_name='failed', to='api.user')),
                ('users_that_succeeded', models.ManyToManyField(blank=True, related_name='succeeded', to='api.user')),
            ],
        ),
    ]
